src/base.ts

```diff
@@ -0,0 +1,290 @@
+	/** Property to sort by (e.g., 'file.name', 'status', 'note.date') */
+	property: string;
+	/** Sort direction */
+	direction: 'ASC' | 'DESC';
+}
+
+/**
+ * Group by configuration for a view.
+ */
+export interface BaseGroupByConfig {
+	/** Property to group by */
+	property: string;
+	/** Sort direction for groups */
+	direction: 'ASC' | 'DESC';
+}
+
+/**
+ * Represents a view in a Base file.
+ */
+export interface BaseView {
+	/** Type of view (table, cards, list, etc.) */
+	type: string;
+	/** Name of the view */
+	name: string;
+	/** Optional: View-specific filters (same format as top-level filters) */
+	filters?: string[];
+	/** Optional: Filter logic for view-specific filters */
+	filterLogic?: 'and' | 'or';
+	/** Optional: Column order for table views */
+	order?: string[];
+	/** Optional: Sort configuration */
+	sort?: BaseSortConfig[];
+	/** Optional: Group by configuration */
+	groupBy?: BaseGroupByConfig;
+	/** Optional: Limit number of results */
+	limit?: number;
+	/** Optional: Summaries mapping property to summary key (e.g., {'price': 'sum', 'quantity': 'average'}) */
+	summaries?: Record<string, string>;
+	/** Optional: Additional view-specific data */
+	data?: Record<string, any>;
+}
+
+/**
+ * Property display configuration.
+ */
+export interface BasePropertyConfig {
+	/** Display name for the property */
+	displayName: string;
+}
+
+/**
+ * Options for generating a Base file.
+ */
+export interface BaseOptions {
+	/** Optional: Filter expressions (e.g., ['file.folder == "Import"', 'file.tags.contains("#todo")']) */
+	filters?: string[];
+	/** Optional: Filter logic - 'and' or 'or' (defaults to 'and') */
+	filterLogic?: 'and' | 'or';
+	/** Optional: Formula definitions (e.g., {'full_name': 'firstName + " " + lastName'}) */
+	formulas?: Record<string, string>;
+	/** Optional: Property configurations (e.g., {'status': {displayName: 'Status'}}) */
+	properties?: Record<string, BasePropertyConfig>;
+	/** Optional: Summary formula definitions (e.g., {'total': 'sum(price)'}) */
+	summaries?: Record<string, string>;
+	/** Views to include in the Base file */
+	views: BaseView[];
+	/** Optional: Folder path for new items created from this base */
+	newItemFolder?: string;
+	/** Optional: Template file path for new items */
+	newItemTemplate?: string;
+}
+
+/**
+ * Generates the content for an Obsidian Base file.
+ * 
+ * @param options - Configuration for the Base file
+ * @returns Base file content in YAML format
+ * 
+ * @example
+ * ```ts
+ * const content = generateBaseContent({
+ *   filters: ['file.folder == "CSV import"'],
+ *   formulas: { full_name: 'firstName + " " + lastName' },
+ *   properties: { status: { displayName: 'Status' } },
+ *   views: [{
+ *     type: 'table',
+ *     name: 'Table',
+ *     order: ['file.name', 'title', 'date', 'category'],
+ *     sort: [{ property: 'file.name', direction: 'ASC' }]
+ *   }]
+ * });
+ * ```
+ */
+export function generateBaseContent(options: BaseOptions): string {
```

Comment:
Why does this function exist instead of using a yaml library to serialize it? That's what we're doing in the app. If we used the same library then we would know the same input produces the same result.